internal/codespaces/sshconfig/sshconfig.go

```diff
@@ -0,0 +1,56 @@
+package sshconfig
+
+import (
+	"io"
+	"strings"
+	"text/template"
+
+	"github.com/MakeNowJust/heredoc"
+)
+
+// Config contains values needed to write an OpenSSH host configuration for a
+// single codespace. For example:
+//
+// Host {{Name}}.{{EscapedRef}
+//
+//	User {{SSHUser}
+//	ProxyCommand {{GHExec}} cs ssh -c {{Name}} --stdio
+//
+// EscapedRef is included in the name to help distinguish between codespaces
+// when tab-completing ssh hostnames. '/' characters in EscapedRef are flattened
+// to '-' to prevent problems with tab completion or when the hostname appears
+// in ControlMaster socket paths.
+type Config struct {
```

Comment:
This was extracted from the ssh command and is used for `gh cs ssh --config` command. It was previously declared inline on the command for each codespace.